### ISSUE TITLE
valid-v-on does not accept an anonymous function

### ISSUE THREAD

**Issue by bdiz** [vuejs/eslint-plugin-vue#716]
**Tell us about your environment**	
* **ESLint version: 4.19.1**	
* **eslint-plugin-vue version: 4.7.1**	
* **Node version: 10.5.0**	

**Please show your full configuration:**

```json
{
  "root": true,
  "parserOptions": {
    "ecmaVersion": 2017
  },  
  "env": {
    "node": true
  },  
  "extends": [
    "plugin:vue/essential",
    "eslint:recommended"
  ],  
  "rules": {
    "indent": ["error", 2], 
    "linebreak-style": ["error", "unix"],
    "quotes": ["error", "double"],
    "semi": ["error", "always"],

    "comma-dangle": ["error", "never"]
  },  
  "globals": {
    "nw": false,
    "GIT_VERSION": false,
    "APP_VERSION": false
  }
}
```

**What did you do?**	

```vue
<template>
<!-- ... -->
        <v-checkbox
          label="Includes a master"
          hide-details
          :input-value="uvc.hasMaster"
          @change="function(hasMaster) { uvc.update({ hasMaster }) }"
        />
<!-- ... -->
</template>
```

**What did you expect to happen?**
I expected eslint vue/valid-v-on checks to pass.

**What actually happened?**
```
error: 'v-on' directives require that attribute value or verb modifiers (vue/valid-v-on) at src/views/uvcs/edit.vue:155:11:
  153 |           hide-details
  154 |           :input-value="uvc.hasMaster"
> 155 |           @change="function(hasMaster) { uvc.update({ hasMaster }) }"
      |           ^
  156 |         />
  157 |         <vue-slide-up-down :active="uvc.hasMaster">
  158 |           <code-area
```

**Comment by ota-meshi** [comment]
Thank you for this issue.

It was reproduced on [vue-eslint-demo](
https://mysticatea.github.io/vue-eslint-demo/#eJyNVrtu5TYQ/RVF1S5g2kBKwwkCpEqRKkEqN5Q0krimSC0f97GG/z2H5BVFSdfAFip4zpCcx5mh3utWd1Q/1y+OpllyR7+/qqp6+YWx6vHxsWIsrf/7c6T2rdGXsKoqyRuSv73Wf6lW+o5sxauJW0fmtU4Go+iIdeS4kDYhz0LN3rETl56w05/ax5Hbvze7/mhHroZA9161Tmj1Jdt8rd6rsMnPHbz88l5lpvr4Wn2kI57g7db3l6ccWP1QGy/J1s/viFpZZ3zrtGHWz2Tq518f6h6rThiKV0dkIIcbmCHnTUKUZi23IbhWcsODpc2E5NYyfGJYjfU0c0NM0cB+kNErrrqDqbJaUrl0d0wcVy5uF9nNxDijJVwd6LKAHTV+GG7RxbUkRyjCiiB4xs1gN0AKZKKpIbNl3uhaAlK0SG3MyIIi3e7KUEnD25C7eNaWnHnIqtqAeXHZhUwXZzhrNDLDVbjJbRlLk1iQnkvpRqP9MGYIStqdOEjdcLkDhVJw9l5VhYJoRZcyO2fUYO1hzM6jcGRn3uYUTOJCHYsQ5IA6O97k4xSdmb1OjZYLoptviEvKbKJbxzNrKDlFKwA/bqcvmCXZ7wKCgbGhtoavJXOjsKwhCJ0K3YPwqqN+XSBQdAGCmLx0QgpFK2eIo7qNLCDLe2K9UAjiukOhe17q1CtvcW6cIHYHQpcrhJAIIiTb8jldhTzBKWZHbRwGRRdBQ989WpZdBcmEYCcTVvF0Zaqdu86kU3wnT0/QnxGNh3LH6zySWj3csJZp091yFAj08kRovtuMOFHJzFoFTvEJ1yu2jJ2gWKGGbDm6SaIndABZgx55Ixc0kXP8qVGo+OEkiuq6tW9GBaqp3Bb77rWjnV2Uze2ezHyzl5B9u5YjoBNHY66JgXTYxuUsFJb8khRTFcZSTMsuwsljjrVjSBVGwaxlLMEhxpjN2WhcF4bKNpVQCbdXtDfSHQrgg2BvxqIINU3H3sdsnliY8icm+pLfzrYVvM23NfCSjyGXjXjDQ+OFq8gYbUrCkCVzgpP7myymJeBVRXjj+MYgvKfU95Cd/Ylwl9GDSQh9oEMw7sJsiHtzMOWOLFe4dheHi50+b6mLg9ucLbUsyVtH54jsHXKjr9TysTDsLNyYClXSp6isjMTWXHKxuyNkJXXiTjLLtFgzLeyBxCTkKG3M7YGMR4dxwop/iM+tjqcbtOz2l6JkkzzLImQmDrJPfAtn3dPFeoxtjZjdfjKE5Ej6+baN70cx3TKRaocOpgue/QyfWIMbmXVXSQUYOn0LpSm9iM1o7XZcOukAYnjxtwOKp4XKDi/xA1jKbME2SlvAOwdO+Hs+GhZvyQq1x5tnc8Twup0PYGg1gB8PdarfP+JH2pmW/17DC1mnf5CHOj78iKluwiOLFxQldPXH/3vZGGc=).

Perhaps a parse error occurs in [vue-eslint-parser](https://github.com/mysticatea/vue-eslint-parser).

I think the parser can not parse the code that starts with `function`.
Vue seems to process correctly.
https://github.com/vuejs/vue/blob/master/src/compiler/codegen/events.js#L3

**Comment by 10suleyman01** [comment]
<b>zsdf</b><hr> @ota-meshi
